### PR TITLE
feat(DENG-3183): add views for deriving firefox fenix retention metrics

### PR DESCRIPTION
**Pull request by kik-kik**
# feat(DENG-3183): add fenix retention queries

This change adds views as part of re-implementation of supporting metrics for our DAU KPI.

The following new views are being introduced:

    retention_clients (view) - calculates each clients retention using a 28 day rolling window.
    retention (table + view) - Aggregation of the retention_clients table.

These tables and views should result in the following tables and views no longer being needed:

    funnel_retention_clients
    funnel_retention_week_4
    funnel_retention_clients_week_2_v1
    funnel_retention_clients_week_4_v1
    funnel_retention_week_4_v1

---

Miro board showing how the data lineage should look like:
https://miro.com/app/board/uXjVNkSmopg=/?share_link_id=21088854196

---

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3476)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/fenix/retention_clients/view.sql [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+attribution AS (
+  SELECT
+    client_id,
+    sample_id,
+    channel AS normalized_channel,
+    adjust_ad_group,
+    adjust_campaign,
+    adjust_creative,
+    adjust_network,
+    install_source,
```
Comment (bani) [inline sql/moz-fx-data-shared-prod/fenix/retention_clients/view.sql:26]: I received a request to have all the different stores on this field. As it is right now, firefox_android_clients is grouping values as "Other"
`fenix_derived.firefox_android_clients_v1` contains the info that we need.

File: sql/moz-fx-data-shared-prod/fenix/retention_clients/view.sql [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  clients_daily.first_seen_date,
+  clients_daily.client_id,
+  clients_daily.sample_id,
+  clients_daily.normalized_channel,
+  clients_daily.country,
+  clients_daily.app_display_version,
+  clients_daily.locale,
+  attribution.adjust_ad_group,
+  attribution.adjust_campaign,
+  attribution.adjust_creative,
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/fenix/retention_clients/view.sql:42]: Include `isp` in the final result of the view.

File: sql/moz-fx-data-shared-prod/fenix/retention_clients/view.sql [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  retention_seen.day_27.active_on_metric_date
+  AND retention_seen.day_27.active_in_week_3 AS ping_sent_week_4,
+  -- activity retention
+  retention_active.day_27.active_on_metric_date AS active_metric_date,
+  retention_active.day_27.active_on_metric_date
+  AND retention_active.day_27.active_in_week_3 AS retained_week_4,
+  -- new client retention
+  clients_daily.is_new_profile AS new_client_metric_date,
+  clients_daily.is_new_profile
+  AND retention_active.day_27.active_in_week_3 AS retained_week_4_new_client,
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/fenix/retention_clients/view.sql:62]: rename all `new_clients` to `new_profiles`

File: sql/moz-fx-data-shared-prod/fenix_derived/retention_v1/query.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
+SELECT
+  metric_date,
+  first_seen_date,
+  app_name,
+  normalized_channel,
+  country,
+  app_display_version,
+  locale,
+  adjust_ad_group,
+  adjust_campaign,
```
[the remark this answers is not in the document]
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/retention_v1/query.sql:10]: I believe we made this change together. Marking as resolved, if you think more changes are required please feel free to reopen or create a new conversation here.

**Review by bani — APPROVED** [review]

**Pull request merged**